### ISSUE TITLE
welcome screen with the ascii cup interferes with rainbow delimiters

### ISSUE THREAD

**Issue by bigos** [lfe/lfe#215]
I like to have nicely colored brackets, sot that I can find matching bracket at a glance without having to move corsor. I have written a script to correct emacs themes so that I can always have resonable indication of matching bracket. To my disappointment it does not work in inferior-lfe unlesst you comment out the ascii cup.

![lfe](https://cloud.githubusercontent.com/assets/392498/14177945/7da5891a-f750-11e5-9cc5-a6a9a471dcb7.png)

**Comment by oubiwann** [comment]
Huh.  That's an interesting dilemma.  

Maybe we could make the vertical bar a matching set of double semi-colons ...  that's rather ugly, though.

Maybe a better approach would be to support a `-no-banner` cmd line option that inferior-lfe would use?

**Comment by bigos** [comment]
I guess -no-banner could be better as long as it is easily to find and documented,

**Comment by mqsoh** [comment]
Might be a cool challenge to try and balance the parens in the ascii art. :smile:

I didn't get the cup handle quite right.

```
 ..-~.~_~---..
(      \\     )    |   A Lisp-2+ on the Erlang VM
|`-.._/_\\_.-';_   |   Type (help) for usage info.
|         g (_  )  |
|        n    | |  |   Docs: http://docs.lfe.io/
(       a    _) |  |   Source: http://github.com/rvirding/lfe
 \     l    (___)  |
  \   r     /      |   LFE v1.0 (abort with ^G)
   `-E___.-'
```

**Comment by oubiwann** [comment]
@mqsoh Ha! Well-done :-)

**Comment by bigos** [comment]
it works

![lfe cropped](https://cloud.githubusercontent.com/assets/392498/14181849/b90e9f72-f760-11e5-80c5-1caedcd9c00a.png)

**Comment by oubiwann** [comment]
Everything we've talked about so far is essentially a work-around and doesn't address the root issue: why is inferior-lfe attempting to parse the banner?

I know when using `rlwrap` (readline-wrap), one can provide a regular expression that tells `rlwrap` when to perform prompt-oriented actions upon startup, and what to ignore. Does inferior-lfe provide a similar facility?

For instance, could we tell it not to start interpreting input until it sees `\n>\s`?

**Comment by oubiwann** [comment]
cc'ing @yurrriq, since he's done a lot with inferior-lfe ...

**Comment by yurrriq** [comment]
Thanks for reporting, @bigos. I highlight trailing whitespace in red and there are a couple instances of that too. My bandwidth is running low atm, but I'll add this to my list, in case no one beats me to it. Oddly, `paredit-mode` (one of my custom hooks) works for me in `inferior-lfe`...

**Comment by yurrriq** [comment]
:+1: to `no-banner`. As a silly workaround, `C-c M-o` clears the buffer, so you could call `inferior-lfe-clear-buffer` before enabling `rainbow-delimiters-mode`.

---

``` elisp
(add-hook 'inferior-lfe-mode-hook
          (lambda ()
            (inferior-lfe-clear-buffer)
            (paredit-mode)
            (rainbow-delimiters-mode)))
```

**Comment by yurrriq** [comment]
While we're at it, I'd love to hook into the shell's autocomplete functionality.

**Comment by rvirding** [comment]
I think the reason that inferior-lfe is parsing the banner because that is
what is does for output into an inferior lfe buffer. Not much to do about
that I think.

**Comment by bigos** [comment]
temporary solution

https://github.com/bigos/lfe/commit/fa82f8898a9786fb1734593b3093c26a84fe886b

**Comment by rvirding** [comment]
Wouldn't the simplest change just be

```
~/lfe/lfe-test$ bin/lfe
Erlang/OTP 18 [erts-7.0] [source-4d83b58] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

   ..-~.~_~---..
  (      \\     )    |   A Lisp-2+ on the Erlang VM
  |`-.._/_\\_.-';    |   Type (help) for usage info.
  |         g |_ \   |   
  |        n    | |  |   Docs: http://docs.lfe.io/ 
  |       a    / /   |   Source: http://github.com/rvirding/lfe
   \     l    |_/    |   
    \   r     /      |   LFE v1.1-dev (abort with ^G)
     `-E___.-'

> 
```

In colour of course. Not quite as nice as the original but with balanced parens.

Robert

**Comment by yurrriq** [comment]
+1 for minor tweaks and a no-banner option.

**Comment by bigos** [comment]
I did it to tweak my emacs startup file without the banner getting in the way. I can have another go at the fix preserving the banner tomorrow, unless somebody beats me to it.

**Comment by bigos** [comment]
fixed in my fork:
https://github.com/bigos/lfe/commit/36ebe50ff219d7cac7eef20eb7c9e5f92bd87117

**Comment by yurrriq** [comment]
**Note**: Robert has addressed this with c6b3e06.
